packages/billing/stripe-webhook-handlers/subscription-deleted.ts

```diff
@@ -71,9 +73,43 @@ export async function handleSubscriptionDeleted(
 
     // This happens when the payment method fails for a couple of times and then Stripe subscription gets expired.
     if (newStatus === 'expired') {
-      // TODO: Put the user back on the free plan (by calling Stripe API). Will be handled in CS-7466
+      await subcribeUserToFreePlan(dbAdapter, event.data.object.customer);
     }
 
     await markStripeEventAsProcessed(dbAdapter, event.id);
   });
 }
+
+async function subcribeUserToFreePlan(
+  dbAdapter: DBAdapter,
+  stripeCustomerId: string,
+) {
+  try {
+    let stripe = getStripe();
+    let freePlan = await getPlanByMonthlyPrice(dbAdapter, 0);
+    if (!freePlan) {
+      console.error('Free plan is not found');
+      return;
+    }
+    let prices = await stripe.prices.list({
+      product: freePlan.stripePlanId,
+      active: true,
+    });
+    if (!prices.data[0]) {
+      console.error('No price found for free plan');
+      return;
+    }
+
+    await stripe.subscriptions.create({
+      customer: stripeCustomerId,
+      items: [
+        {
+          price: prices.data[0].id,
+        },
+      ],
+      payment_behavior: 'error_if_incomplete',
```

Comment:
Add a comment why this behaviour is needed. Please also add more comments on how this works after a subscription is created, i.e. describing that the the invoice.payment_succeeded webhook will take care of making a new subscription, add plan allowance etc.

Reply:
Have you tested this locally using Stripe's sandbox? Does the subscription get created successfully?